octopoes/octopoes/repositories/ooi_repository.py

```diff
@@ -590,16 +590,26 @@ def count_findings_by_severity(self, valid_time: datetime) -> Counter:
 
         query = """
             {:query {
-                :find [(pull ?finding_type [*]) (count ?finding)]
+                :find [?finding_type (pull ?finding_type [*]) (count ?finding)]
                 :where [
                     [?finding :Finding/finding_type ?finding_type]
                     (not-join [?finding] [?muted_finding :MutedFinding/finding ?finding])
                     ]}}
         """
 
-        for finding_type, finding_count in self.session.client.query(str(query), valid_time=valid_time):
-            ft = cast(FindingType, self.deserialize(finding_type))
-            severity = ft.risk_severity or RiskLevelSeverity.PENDING
+        for finding_type_name, finding_type_object, finding_count in self.session.client.query(
+            str(query), valid_time=valid_time
+        ):
+            if not finding_type_object:
+                logger.warning(
```

Comment:
Another route would be to just add them? FindingTypes are relatively simple objects I'd say, and, adding them could be the only clear way out of these errors for the user anyway.

Reply:
Findings without finding types indicate that something went wrong somewhere. I don't think it is a good idea to try to fix those data issues by adding objects in this view instead of fixing the root cause. Doing that would also mean that the data in the graph will depend on whether a user visited a certain page or not which I don't think is a good design.

What we could do is change the way finding types are created and automatically create a finding type if a finding is created instead of making every plugin responsible for also creating the finding type. But that's something for another ticket.